packages/yarnpkg-parsers/sources/syml.ts

```diff
@@ -141,10 +140,22 @@ function parseViaJsYaml(source: string) {
   if (LEGACY_REGEXP.test(source))
     return parseViaPeg(source);
 
-  const value = safeLoad(source, {
-    schema: FAILSAFE_SCHEMA,
-    json: true,
-  });
+  // js-yaml@3 doesn't parse `key: true` as a boolean, but as a string.
+  // as far as I tested it makes the lock file invalid after install.
+  const reviver = (key: string, value: unknown) => {
```

Comment:
It's more a difference... The yaml parser parse the native value, js-yaml keep a string. Without this it broke the lock file, but I guess a refactor can be done in the stringify too, thus keeping what yaml does.